windows/CodePush.cs

```diff
@@ -0,0 +1,668 @@
+            try
+            {
+                devBundleCacheFile = await ApplicationData.Current.LocalFolder.GetFileAsync(REACT_DEV_BUNDLE_CACHE_FILE_NAME);
+            }
+            catch (FileNotFoundException)
+            {
+            }
+
+            if (devBundleCacheFile != null)
+            {
+                await devBundleCacheFile.DeleteAsync();
+            }
+        }
+
+        private async Task<long> GetBinaryResourcesModifiedTime()
+        {
+            StorageFile assetJSBundleFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(ASSETS_BUNDLE_PREFIX + assetsBundleFileName));
+            BasicProperties fileProperties = await assetJSBundleFile.GetBasicPropertiesAsync();
+            return fileProperties.DateModified.ToUnixTimeMilliseconds();
+        }
+
+        public string GetJavaScriptBundleFile()
+        {
+            return GetJavaScriptBundleFile(DEFAULT_JS_BUNDLE_NAME);
+        }
+
+        public string GetJavaScriptBundleFile(string assetsBundleFileName)
+        {
+            if (currentInstance == null)
+            {
+                throw new CodePushNotInitializedException("A CodePush instance has not been created yet. Have you added it to your app's list of ReactPackages?");
+            }
+
+            return currentInstance.GetJavaScriptBundleFileAsync(assetsBundleFileName).Result;
+        }
+
+        public async Task<string> GetJavaScriptBundleFileAsync(string assetsBundleFileName)
+        {
+            this.assetsBundleFileName = assetsBundleFileName;
+            string binaryJsBundleUrl = ASSETS_BUNDLE_PREFIX + assetsBundleFileName;
+            long binaryResourcesModifiedTime = await GetBinaryResourcesModifiedTime();
+            StorageFile packageFile = await codePushPackage.GetCurrentPackageBundle(this.assetsBundleFileName);
+            if (packageFile == null)
+            {
+                // There has not been any downloaded updates.
+                CodePushUtils.logBundleUrl(binaryJsBundleUrl);
+                isRunningBinaryVersion = true;
+                return binaryJsBundleUrl;
+            }
+
+            JObject packageMetadata = await codePushPackage.GetCurrentPackage();
+            long? binaryModifiedDateDuringPackageInstall = null;
+            string binaryModifiedDateDuringPackageInstallString = (string)packageMetadata[BINARY_MODIFIED_TIME_KEY];
+            if (binaryModifiedDateDuringPackageInstallString != null)
+            {
+                binaryModifiedDateDuringPackageInstall = long.Parse(binaryModifiedDateDuringPackageInstallString);
+            }
+
+            string packageAppVersion = (string)packageMetadata["appVersion"];
+
+            // TODO: test configuration
+            if (binaryModifiedDateDuringPackageInstall != null &&
+                    binaryModifiedDateDuringPackageInstall == binaryResourcesModifiedTime &&
+                    (IsUsingTestConfiguration() || appVersion.Equals(packageAppVersion)))
```

Comment:
It was a hack for the iOS version of the tests, where it was not easy to test changes the info.plist app version in the XCTest framework. I agree this should be removed in the windows implementation.